File: api/typescript/src/engine/lib/kurtosis_context/log_line_filter.ts
import {LogLineOperator} from "./log_line_operator";

const DEFAULT_OPERATOR: LogLineOperator = LogLineOperator.DoesContainText;

export class LogLineFilter {

    private operator: LogLineOperator = DEFAULT_OPERATOR;
    private textPattern: string = "";

    public getOperator(): LogLineOperator {
        return this.operator;
    }

    public getTextPattern(): string {
        return this.textPattern;
    }

    public static NewDoesContainTextLogLineFilter(text: string): LogLineFilter {
        const operator: LogLineOperator = LogLineOperator.DoesContainText;
        return this.newLogLineFilter(operator, text);
    }

    public static NewDoesNotContainTextLogLineFilter(text: string): LogLineFilter {
        const operator: LogLineOperator = LogLineOperator.DoesNotContainText;
        return this.newLogLineFilter(operator, text);
    }

    public static NewDoesContainMatchRegexLogLineFilter(regex: string): LogLineFilter {
        const operator: LogLineOperator = LogLineOperator.DoesContainMatchRegex;
        return this.newLogLineFilter(operator, regex);
    }

    public static NewDoesNotContainMatchRegexLogLineFilter(regex: string): LogLineFilter {
        const operator: LogLineOperator = LogLineOperator.DoesNotContainMatchRegex;
        return this.newLogLineFilter(operator, regex);
    }

    private static newLogLineFilter(operator: LogLineOperator, textPattern: string): LogLineFilter {
        const filter: LogLineFilter = new LogLineFilter();
        filter.operator = operator;
        filter.textPattern = textPattern;
        return filter;
    }
}

File: api/golang/engine/lib/kurtosis_context/log_line_filter.go
package kurtosis_context

type LogLineFilter struct {
	operator    logLineOperator
	textPattern string
}

func NewDoesContainTextLogLineFilter(text string) *LogLineFilter {
	return &LogLineFilter{operator: logLineOperator_DoesContainText, textPattern: text}
}

func NewDoesNotContainTextLogLineFilter(text string) *LogLineFilter {
	return &LogLineFilter{operator: logLineOperator_DoesNotContainText, textPattern: text}
}

func NewDoesContainMatchRegexLogLineFilter(regex string) *LogLineFilter {
	return &LogLineFilter{operator: logLineOperator_DoesContainMatchRegex, textPattern: regex}
}

func NewDoesNotContainMatchRegexLogLineFilter(regex string) *LogLineFilter {
	return &LogLineFilter{operator: logLineOperator_DoesNotContainMatchRegex, textPattern: regex}
}
